gitops/snapshot.go

```diff
@@ -998,29 +1033,58 @@ func IsContextValidForSnapshot(scenarioContextName string, snapshot *application
 	return false
 }
 
-// IsScenarioApplicableToSnapshotsContext checks the contexts list for a given IntegrationTestScenario and
-// compares it against the Snapshot to determine if the scenario applies to it
-func IsScenarioApplicableToSnapshotsContext(scenario *v1beta2.IntegrationTestScenario, snapshot *applicationapiv1alpha1.Snapshot) bool {
+// IsContextValidForBuildPLR checks the context and compares it against the snapshot which will be created for build PLR to determine if it applies
+func IsContextValidForBuildPLR(scenarioContextName string, plr *tektonv1.PipelineRun) bool {
+	// `application` context is supported for backwards-compatibility and considered the same as `all`
+	if scenarioContextName == "application" || scenarioContextName == "all" {
+		return true
+	} else if scenarioContextName == "component" {
+		return true
+	} else if strings.HasPrefix(scenarioContextName, "component_") {
+		componentName := strings.TrimPrefix(scenarioContextName, "component_")
+		if metadata.HasLabelWithValue(plr, SnapshotComponentLabel, componentName) {
+			return true
+		}
+	} else if scenarioContextName == "pull_request" && !tekton.IsPLRCreatedByPACPushEvent(plr) {
+		return true
+	}
+	return false
+}
+
+// IsScenarioApplicableToObjectContext checks the contexts list for a given IntegrationTestScenario and
+// compares it against the Snapshot/Build pipelinerun to determine if the scenario applies to it
+func IsScenarioApplicableToObjectContext(scenario *v1beta2.IntegrationTestScenario, object metav1.Object) bool {
 	// If the contexts list is empty, we assume that the scenario applies to all contexts by default
 	if len(scenario.Spec.Contexts) == 0 {
 		return true
 	}
-	for _, scenarioContext := range scenario.Spec.Contexts {
-		scenarioContext := scenarioContext //G601
-		if IsContextValidForSnapshot(scenarioContext.Name, snapshot) {
-			return true
+	if snapshot, ok := object.(*applicationapiv1alpha1.Snapshot); ok {
```

Comment:
Could we do something like this (untested):
```
                for _, scenarioContext := range scenario.Spec.Contexts {
			scenarioContext := scenarioContext //G601
                        if snapshot, ok := object.(*applicationapiv1alpha1.Snapshot); ok {
                            return IsContextValidForSnapshot(scenarioContext.Name, snapshot)
                        }
                        if plr, ok := object.(*tektonv1.PipelineRun); ok {
			    return IsContextValidForBuildPLR(scenarioContext.Name, plr)
			}
		}
```

Reply:
Yes, thanks!